opsgenie/data_source_opsgenie_service.go

```diff
@@ -0,0 +1,89 @@
+package opsgenie
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/service"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func dataSourceOpsGenieService() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceOpsGenieServiceRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateOpsGenieServiceName,
+			},
+			"team_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateOpsGenieServiceTeamId,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateOpsGenieServiceDescription,
+			},
+		},
+	}
+}
+
+func dataSourceOpsGenieServiceRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := service.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Reading OpsGenie service '%s'", name)
+
+	res, err := client.List(context.Background(), &service.ListRequest{
+		Limit:  10,
```

Comment:
why do we use limit 10? 
the default value is 20 and max is 100, we can use max value for the avoid request throttling
also why do call this list two different place? we can make this inside for loop below

Reply:
Changed limit to 100
Calling API once